source/security/authentication.txt

```diff
@@ -0,0 +1,403 @@
+             'authMechanism' => 'SCRAM-SHA-1',
+         ];
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            $options,
+         );
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<username>:<password>@<hostname>:<port>/?authSource=admin&authMechanism=SCRAM-SHA-1';
+         $client = new Client($uri);
+
+.. _php-mongodb-x509:
+
+MONGODB-X509
+------------
+
+If you enable TLS, during the TLS handshake, the {+php-library+} can present an X.509
+client certificate to MongoDB to prove its identity. The MONGODB-X509 authentication
+mechanism uses this certificate to authenticate the client.
+
+To authenticate with this mechanism, set the following connection options:
+
+- ``tls``: Set to ``true``.
+- ``tlsCertificateKeyFile``: The file path of the ``.pem`` file that contains your
+  client certificate and private key.
+- ``authMechanism``: Set to ``'MONGODB-X509'``.
+
+You can set these options in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through parameters in your connection string.
+
+.. tabs::
+
+   .. tab:: MongoDB\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $options = [
+             'tls' => true,
+             'tlsCertificateKeyFile' => '<file path>',
+             'authMechanism' => 'MONGODB-X509',
+         ];
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            $options,
+         );
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<hostname>:<port>/?tls=true&tlsCertificateKeyFile=<file path>&authMechanism=MONGODB-X509';
+         $client = new Client($uri);
+
+.. _php-mongodb-aws:
+
+MONGODB-AWS
+-----------
+
+.. important::
+
+   The MONGODB-AWS authentication mechanism requires MongoDB v4.4 or later.
+
+The MONGODB-AWS authentication mechanism uses AWS IAM (Amazon Web Services Identity and
+Access Management) or AWS Lambda credentials to authenticate your application. To
+authenticate using this mechanism, first create a user with an associated Amazon Resource
+Name (ARN) on the ``$external`` database, then specify ``'MONGODB-AWS'`` as the value of
+the ``authMechanism`` connection option.
+
+When you use the MONGODB-AWS mechanism, the {+php-library+} tries to retrieve AWS
+credentials from the following sources, in the order listed:
+
+1. Options parameter passed to the ``MongoDB\Client`` constructor or parameters in the
+   connection URI
+#. Environment variables
+#. AWS EKS ``AssumeRoleWithWebIdentity`` request
+#. ECS container metadata
+#. EC2 instance metadata
+
+The following sections describe how to use the {+php-library+} to retrieve credentials from
+these sources and use them to authenticate your application.
+
+.. _php-mongodb-aws-credentials:
+
+``MongoDB\Client`` Credentials
```

Comment:
The Style Guide technically prohibits using monospace in titles and headings (though I think this is a bad rule)

```suggestion
MongoDB\Client Credentials
```

Reply:
agreed (but fixed)